arangodb-net-standard/Serialization/IContentSerialization.cs

```diff
@@ -0,0 +1,11 @@
+﻿using System.IO;
+
+namespace ArangoDBNetStandard.Serialization
+{
+    public interface IContentSerialization
+    {
+        T DeserializeJsonFromStream<T>(Stream stream);
+
+        string SerializeToJson<T>(T item, bool useCamelCasePropertyNames, bool ignoreNullValues);
```

Comment:
I realize that this shouldn't mention `JSON` if want to use this interface as a way to provide different serialization formats, e.g. VelocyPack.

So either we consider this interface as a JSON serialization interface only and we would handle different serialization format with different interface at the API client level (seems only 2 with JSON and VelocyPack).
Or we consider it as a more general serialization layer that can be used to provide different serialization formats (JSON, VelocyPack).

I am in favor of the second approach. Though `useCamelCasePropertyNames` and `ignoreNullValues` are really JSON specific.

Reply:
Some new efficient serializers also support serializing to a JSON string, encoded as UTF-8 bytes, e.g. [System.Text.Json.JsonSerializer](https://docs.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializer?view=netcore-3.0) or [Utf8Json](https://github.com/neuecc/Utf8Json). Same for [VelocyPack](https://github.com/arangodb/velocypack/blob/master/VelocyPack.md) from what I understand.

I wonder if we should work with a `byte[] Serialize<T>(T item)` method instead.

Reply:
In the java driver, it seems they manage the serialization formats separately, so JSON and VPack.
https://github.com/arangodb/arangodb-java-driver/blob/master/src/main/java/com/arangodb/Protocol.java

Reply:
I had a discussion with @rossmills99 
We agreed to provide an interface that uses `byte[]` or `Stream` for the `Serialize<T>` method.
`IApiClientTransport` layer will also be updated to accept `byte[]` or `Stream`.
Default serialization implementation will still be provided with Json.NET.

Reply:
Finally I went with `byte[]` because the constructor for `ByteArrayContent` accept `byte[]`. We can move to `Stream` if necessary in future.

Reply:
I've opened #153 and I wonder whether we should define a specific model here to help with getting the content-type. Example:

```c#
// Or Payload?
public class MessageContent
{
public byte[] Content;

public string ContentType;
}
```

The property `ContentType` would be set to "application/json" in JSON serializer implementation and "application/vpack" in VelocyPack implementation.

Another way to solve this problem:
- Define a `ContentType` property at the class level that implementation would set. Then the API client can use `_serializer.ContentType` when passing the content to the transport.

Reply:
Would it make sense for the transport implementation to set ContentType headers? In the case of HttpClient, seems like it could be set as a default header for all requests.

Reply:
It would make sense but the transport implementation would still need to know what format the data was serialized to. `application/json` or `application/vpack`.